pytorch_lightning/accelerators/gpu.py

```diff
@@ -52,3 +54,14 @@ def set_nvidia_flags(local_rank: int) -> None:
         all_gpu_ids = ",".join([str(x) for x in range(torch.cuda.device_count())])
         devices = os.getenv("CUDA_VISIBLE_DEVICES", all_gpu_ids)
         _log.info(f"LOCAL_RANK: {local_rank} - CUDA_VISIBLE_DEVICES: [{devices}]")
+
+    def teardown(self) -> None:
+        super().teardown()
+
+        for optimizer in self.optimizers:
+            for k, v in optimizer.state.items():
+                if isinstance(v, Mapping):
+                    optimizer.state[k] = {
+                        n: value.cpu() if isinstance(value, torch.Tensor) else value
+                        for n, value in v.items()
+                    }
```

Comment:
we have a protected helper function 
`_move_optimizer_state` in the accelerator superclass. 
should we repurpose this one?

Reply:
Yes.